src/ui/profile_preferences.ui

```diff
@@ -568,27 +568,109 @@
+         <property name="checked">
+          <bool>false</bool>
+         </property>
+         <layout class="QVBoxLayout" name="verticalLayout_2">
+          <item>
+           <widget class="QLabel" name="label_chooseSpellCheckDictionary">
             <property name="text">
              <string>Choose spell check dictionary:</string>
             </property>
            </widget>
           </item>
-          <item row="0" column="1">
-           <widget class="QListWidget" name="dictList"/>
+          <item>
+           <widget class="QListWidget" name="dictList">
+            <property name="flow">
+             <enum>QListView::TopToBottom</enum>
+            </property>
+            <property name="resizeMode">
+             <enum>QListView::Adjust</enum>
+            </property>
+            <property name="viewMode">
+             <enum>QListView::IconMode</enum>
+            </property>
+            <property name="sortingEnabled">
+             <bool>true</bool>
+            </property>
+           </widget>
           </item>
-          <item row="1" column="0" colspan="2">
-           <widget class="QCheckBox" name="enableSpellCheck">
-            <property name="text">
-             <string>Enable spell check</string>
+          <item>
+           <widget class="QGroupBox" name="groupBox_autoAddWordsFromServerToDictionary">
+            <property name="title">
+             <string>Automatically add words from server</string>
             </property>
-            <property name="checked">
+            <property name="checkable">
              <bool>true</bool>
             </property>
+            <property name="checked">
+             <bool>false</bool>
+            </property>
+            <layout class="QGridLayout" name="gridLayout_autoAddWordsFromServerToDictionary" columnstretch="0,1">
+             <property name="horizontalSpacing">
+              <number>27</number>
+             </property>
+             <item row="0" column="0">
+              <widget class="QWidget" name="widget" native="true">
+               <layout class="QHBoxLayout" name="horizontalLayout_autoAddWordToDictionaryMinLength" stretch="0,1,0">
+                <property name="leftMargin">
+                 <number>0</number>
+                </property>
+                <property name="topMargin">
+                 <number>0</number>
+                </property>
+                <property name="rightMargin">
+                 <number>0</number>
+                </property>
+                <property name="bottomMargin">
+                 <number>0</number>
+                </property>
+                <item alignment="Qt::AlignLeft">
+                 <widget class="QLabel" name="label_autoAddDictionaryWordMinLength_before">
+                  <property name="text">
+                   <string extracomment="This (QLabel) texts is placed before a QSpinBox that displays a number with another QLabel that finishes a sentence">Minimum length</string>
```

Comment:
Please don't interrupt sentences like this
```
Minimum length
4
characters
```

How about 
```
Minimum length:
4
```

Or minimum length in characters. 
Or minimum length of words to add.
Yeah I like the last one best.

Reply:
Yeah, it is not a sane idea :stuck_out_tongue_winking_eye: to split the text like this - it might look pretty in English but it is harder to localize...

... I ***will*** fix this!

Reply:
This part has been nuked anyhow so this is no longer an issue.